File: boj/1920.py
N = int(input())
N_list = list(map(int, input().split()))
M = int(input())
M_list = list(map(int, input().split()))

N_list.sort()


for m in M_list:
    start, end = 0, N-1
    flag = False
    while start <= end:
        mid = (start + end) // 2

        if N_list[mid] > m:
            end = mid-1

        elif N_list[mid] == m:
            flag = True
            break

        else:
            start = mid+1

    if flag == True:
        print(1)
    else:
        print(0)


















# N = int(input())
# N_list = list(map(int, input().split()))

# M = int(input())
# M_list = list(map(int, input().split()))

# for m in M_list:
#     if m in N_list:
#         print(1)
#     else:
#         print(0)

File: boj/1920.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {

    static int[] A;
    static int n, m;

    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        n = Integer.parseInt(br.readLine());
        A = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(A);

        m = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < m; i++) {
            int result = isNumber(Integer.parseInt(st.nextToken()));
            System.out.println(result);
        }
    }

    static int isNumber(int num) {
        int start = 0, end = n - 1;

        while (start <= end) {
            int mid = (start + end) / 2;

            if (A[mid] > num) {
                end = mid - 1;
            } else if (A[mid] < num) {
                start = mid + 1;
            } else {
                return 1;
            }
        }

        return 0;
    }
}